modules/conversantAnalyticsAdapter.js

```diff
@@ -77,33 +88,99 @@ export const cnvrHelper = {
-            break;
-          case AD_RENDER_FAILED:
-            onAdRenderFailed(args);
-            break;
-          case BID_WON:
-            onBidWon(args);
-            break;
-          case BID_TIMEOUT:
-            onBidTimeout(args);
-            break;
-        } // END switch
-      } else {
-        logMessage(CNVR_CONSTANTS.LOG_PREFIX + ' - ' + eventType + ': skipped due to sampling');
-      }// END IF(cnvrHelper.doSample)
+      try {
+        if (cnvrHelper.doSample) {
+          logMessage(CNVR_CONSTANTS.LOG_PREFIX + ' track(): ' + eventType, args);
+          switch (eventType) {
+            case AUCTION_END:
+              onAuctionEnd(args);
+              break;
+            case AD_RENDER_FAILED:
+              onAdRenderFailed(args);
+              break;
+            case BID_WON:
+              onBidWon(args);
+              break;
+            case BID_TIMEOUT:
+              onBidTimeout(args);
+              break;
+            case BIDDER_ERROR:
+              onBidderError(args)
+          } // END switch
+        } else {
+          logMessage(CNVR_CONSTANTS.LOG_PREFIX + ' - ' + eventType + ': skipped due to sampling');
+        }// END IF(cnvrHelper.doSample)
+      } catch (e) {
+        // e = {stack:"...",message:"..."}
+        logError(CNVR_CONSTANTS.LOG_PREFIX + 'Caught error in handling ' + eventType + ' event: ' + e.message);
+        cnvrHelper.sendErrorData(eventType, e);
+      }
     } // END track()
   }
 );
 
 // ================================================== EVENT HANDLERS ===================================================
+
+/**
+ * Handler for BIDDER_ERROR events, tries to capture as much data, save it in cache which is then picked up by
+ * AUCTION_END event and included in that payload. Was not able to see an easy way to get adUnitCode in this event
+ * so not including it for now.
+ * https://docs.prebid.org/dev-docs/bidder-adaptor.html#registering-on-bidder-error
+ * Trigger when the HTTP response status code is not between 200-299 and not equal to 304.
+ {
+    error: XMLHttpRequest,  https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
+    bidderRequest: {  https://docs.prebid.org/dev-docs/bidder-adaptor.html#registering-on-bidder-error
+        {
+            auctionId: "b06c5141-fe8f-4cdf-9d7d-54415490a917",
+            auctionStart: 1579746300522,
+            bidderCode: "myBidderCode",
+            bidderRequestId: "15246a574e859f",
+            bids: [{...}],
+            gdprConsent: {consentString: "BOtmiBKOtmiBKABABAENAFAAAAACeAAA", vendorData: {...}, gdprApplies: true},
+            refererInfo: {
+                canonicalUrl: null,
+                page: "http://mypage.org?pbjs_debug=true",
+                domain: "mypage.org",
+                ref: null,
+                numIframes: 0,
+                reachedTop: true,
+                isAmp: false,
+                stack: ["http://mypage.org?pbjs_debug=true"]
+            }
+        }
+    }
+}
+ */
```

Comment:
Do we need this comment?

Reply:
> Do we need this comment?

I like to have detailed comments, I work on many projects and have little backup so if ever someone came in to work on this project that didn't have my support it would help jump start them much faster.  I've found Prebid.js documentation to be pretty poor and with untyped Javascript having a detailed description of the data we are expecting is really useful.